### PR TITLE
deadpixi-sam: fix on Darwin, bump

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build deadpixi-sam

**Comment by veprbl** [comment]
@GrahamcOfBorg build deadpixi-sam

File: pkgs/applications/editors/deadpixi-sam/default.nix [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
+    rev = "51693780fb1457913389db6634163998f9b775b8";
+    sha256 = "0nfkj93j4bgli4ixbk041nwi14rabk04kqg8krq4mj0044m1qywr";
+  };
 
   postPatch = ''
     substituteInPlace config.mk.def \
       --replace "/usr/include/freetype2" "${freetype.dev}/include/freetype2"
+  '' + stdenv.lib.optionalString stdenv.cc.isClang ''
+    substituteInPlace config.mk.def \
+      --replace gcc clang
```
Comment (LnL7) [inline pkgs/applications/editors/deadpixi-sam/default.nix]: you can use `cc` or `$CC` here, that works in all platforms.

**Review by LnL7 — APPROVED** [review]

**Comment by veprbl** [comment]
@GrahamcOfBorg build deadpixi-sam

**Review by LnL7 — APPROVED** [review]

**Pull request merged**